src/gui/CMakeLists.txt

```diff
@@ -262,6 +262,31 @@ if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -L/usr/local/lib")
 endif()
 
+add_library(${APPLICATION_LIB} ${final_src})
```

Comment:
Make it static and give it a fixed name. Since it'll be static it means that at deploy time nothing will change which is a good thing for us, also won't need to mingle with export macros just yet. :-)